test/models/summary_helper_test.exs

```diff
@@ -0,0 +1,367 @@
+defmodule CoursePlanner.SummaryHelperTest do
+  use CoursePlanner.ModelCase
+
+  import CoursePlanner.Factory
+  alias CoursePlanner.SummaryHelper
+
+  @empty_summary_helper_user_data_response %{terms: [], offered_courses: []}
+
+  test "when user role is unknown" do
+    student = insert(:student)
+    [term1, term2] = insert_list(2, :term)
+    insert_list(4, :offered_course, term: term1)
+    insert(:offered_course, term: term2, students: [student])
+
+    student_data = SummaryHelper.get_term_offered_course_for_user(%{student | role: "Unknown"})
+    assert student_data == @empty_summary_helper_user_data_response
+  end
+
+  describe "Student summary data" do
+    test "when she does not exist" do
+      student = build(:student, id: -1, role: "Students")
```

Comment:
I don't think you need to give a role here, and there is an extra `s` in the end btw